src/wolfio.c

```diff
@@ -2714,4 +2714,382 @@ int wolfSSL_SetIO_LwIP(WOLFSSL* ssl, void* pcb,
+    if (ctx->receive_buffer_size < data_size) {
+        WOLFSSL_MSG("ISO-TP buffer is too small to receive data");
+        return BUFFER_E;
+    }
+    XMEMCPY(ctx->receive_buffer, &ctx->frame.data[1], data_size);
+    return data_size;
+}
+
+static int isotp_receive_multi_frame(struct isotp_wolfssl_ctx *ctx)
+{
+    int ret;
+    word16 data_size;
+    byte delay = 0;
+
+    /* Increase receive timeout for enforced ms delay */
+    if (ctx->receive_delay <= ISOTP_MAX_MS_FRAME_DELAY) {
+        delay = ctx->receive_delay;
+    }
+    /* Still processing first frame.
+     * Full data size is lower nibble of first byte for the most significant
+     * followed by the second byte for the rest. Last 6 bytes are data */
+    data_size = ((ctx->frame.data[0] & 0xf) << 8) + ctx->frame.data[1];
+    XMEMCPY(ctx->receive_buffer, &ctx->frame.data[2], ISOTP_FIRST_FRAME_DATA_SIZE);
+    /* Need to send a flow control packet to either cancel or continue
+     * transmission of data */
+    if (ctx->receive_buffer_size < data_size) {
+        isotp_send_flow_control(ctx, TRUE);
+        WOLFSSL_MSG("ISO-TP buffer is too small to receive data");
+        return BUFFER_E;
+    }
+    isotp_send_flow_control(ctx, FALSE);
+
+    ctx->buf_length = ISOTP_FIRST_FRAME_DATA_SIZE;
+    ctx->buf_ptr = ctx->receive_buffer + ISOTP_FIRST_FRAME_DATA_SIZE;
+    data_size -= ISOTP_FIRST_FRAME_DATA_SIZE;
+    ctx->sequence = 1;
+
+    while(data_size) {
+        enum isotp_frame_type type;
+        byte sequence;
+        byte frame_len;
+        ret = ctx->recv_fn(&ctx->frame, ctx->arg, ISOTP_DEFAULT_TIMEOUT +
+                (delay / 1000));
+        if (ret == 0) {
+            return WOLFSSL_CBIO_ERR_TIMEOUT;
+        }
+        type = ctx->frame.data[0] >> 4;
+        /* Consecutive frames have sequence number as lower nibble */
+        sequence = ctx->frame.data[0] & 0xf;
+        if (type != ISOTP_FRAME_TYPE_CONSECUTIVE) {
+            WOLFSSL_MSG("ISO-TP frames out of sequence");
+            return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+        if (sequence != ctx->sequence) {
+            WOLFSSL_MSG("ISO-TP frames out of sequence");
+            return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+        /* Last 7 bytes or whatever we got after the first byte is data */
+        frame_len = ctx->frame.length - 1;
+        XMEMCPY(ctx->buf_ptr, &ctx->frame.data[1], frame_len);
+        ctx->buf_ptr += frame_len;
+        ctx->buf_length += frame_len;
+        data_size -= frame_len;
+
+        /* Sequence is 0 - 15 (first 0 is implied for first packet */
+        ctx->sequence++;
+        if (ctx->sequence > ISOTP_MAX_SEQUENCE_COUNTER) {
+            ctx->sequence = 0;
+        }
+    }
+    return ctx->buf_length;
+
+}
+
+/* The wolfSSL receive callback, needs to buffer because we need to grab all
+ * incoming data, even if wolfSSL doesn't want it all yet */
+int ISOTP_Receive(WOLFSSL* ssl, char* buf, int sz, void* ctx)
+{
+    (void) ssl;
```

Comment:
The unused should go below the variable declarations, some compilers are picky about this...